src/commonMain/kotlin/fr/acinq/lightning/serialization/v4/Deserialization.kt

```diff
@@ -307,43 +308,60 @@ object Deserialization {
-                    commitTx = readTransactionWithInputInfo() as CommitTx,
-                    htlcTxs = readCollection { readTransactionWithInputInfo() as HtlcTx }.toList(),
-                )
-            },
-            readRight = {
-                LocalCommit(
-                    index = readNumber(),
-                    spec = readCommitmentSpecWithHtlcs(),
-                    publishableTxs = PublishableTxs(
-                        commitTx = readTransactionWithInputInfo() as CommitTx,
-                        htlcTxsAndSigs = readCollection {
-                            HtlcTxAndSigs(
-                                txinfo = readTransactionWithInputInfo() as HtlcTx,
-                                localSig = readByteVector64(),
-                                remoteSig = readByteVector64()
-                            )
-                        }.toList()
-                    )
+    private fun Input.readUnsignedLocalCommitWithHtlcs(): InteractiveTxSigningSession.Companion.UnsignedLocalCommit = InteractiveTxSigningSession.Companion.UnsignedLocalCommit(
+        index = readNumber(),
+        spec = readCommitmentSpecWithHtlcs(),
+        commitTx = readTransactionWithInputInfo() as CommitTx,
+        htlcTxs = readCollection { readTransactionWithInputInfo() as HtlcTx }.toList(),
+    )
+
+    private fun Input.readLocalCommitWithHtlcs(): LocalCommit = LocalCommit(
+        index = readNumber(),
+        spec = readCommitmentSpecWithHtlcs(),
+        publishableTxs = PublishableTxs(
+            commitTx = readTransactionWithInputInfo() as CommitTx,
+            htlcTxsAndSigs = readCollection {
+                HtlcTxAndSigs(
+                    txinfo = readTransactionWithInputInfo() as HtlcTx,
+                    localSig = readByteVector64(),
+                    remoteSig = readByteVector64()
                 )
-            },
+            }.toList()
+        )
+    )
+
+    private fun Input.readLiquidityPurchase(): LiquidityAds.Lease = LiquidityAds.Lease(
+        amount = readNumber().sat,
+        fees = LiquidityAds.LeaseFees(miningFee = readNumber().sat, serviceFee = readNumber().sat),
+        sellerSig = readByteVector64(),
+        witness = LiquidityAds.LeaseWitness(
+            fundingScript = readNBytes(readNumber().toInt())!!.toByteVector(),
+            leaseDuration = readNumber().toInt(),
+            leaseEnd = readNumber().toInt(),
+            maxRelayFeeProportional = readNumber().toInt(),
+            maxRelayFeeBase = readNumber().msat,
         ),
-        remoteCommit = RemoteCommit(
+    )
+
+    private fun Input.readInteractiveTxSigningSession(): InteractiveTxSigningSession {
+        val fundingParams = readInteractiveTxParams()
+        val fundingTxIndex = readNumber()
+        val fundingTx = readSignedSharedTransaction() as PartiallySignedSharedTransaction
+        val (liquidityPurchase, localCommit) = when (val discriminator = read()) {
```

Comment:
```suggestion
        // liquidityPurchase and localCommit are logically independent, this is just a serialization trick for backward compatibility
        val (liquidityPurchase, localCommit) = when (val discriminator = read()) {
```

Reply:
Added in https://github.com/ACINQ/lightning-kmp/pull/561/commits/3541777350fdae4b0ac92cbf55d4e52ae1c200c5